msrestazure/polling/arm_polling.py

```diff
@@ -0,0 +1,417 @@
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import re
+import time
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+from msrest.exceptions import DeserializationError, ClientException
+from msrest.polling import PollingMethod
+
+from ..azure_exceptions import CloudError
+
+
+FINISHED = frozenset(['succeeded', 'canceled', 'failed'])
+FAILED = frozenset(['canceled', 'failed'])
+SUCCEEDED = frozenset(['succeeded'])
+
+
+def finished(status):
+    if hasattr(status, 'value'):
+        status = status.value
+    return str(status).lower() in FINISHED
+
+
+def failed(status):
+    if hasattr(status, 'value'):
+        status = status.value
+    return str(status).lower() in FAILED
+
+
+def succeeded(status):
+    if hasattr(status, 'value'):
+        status = status.value
+    return str(status).lower() in SUCCEEDED
+
+
+class BadStatus(Exception):
+    pass
+
+
+class BadResponse(Exception):
+    pass
+
+
+class OperationFailed(Exception):
+    pass
+
+def _validate(url):
+    """Validate a url.
+
+    :param str url: Polling URL extracted from response header.
+    :raises: ValueError if URL has no scheme or host.
+    """
+    if url is None:
+        return
+    parsed = urlparse(url)
+    if not parsed.scheme or not parsed.netloc:
+        raise ValueError("Invalid URL header")
+
+def get_header_url(response, header_name):
+    """Get a URL from a header requests.
+
+    :param requests.Response response: REST call response.
+    :param str header_name: Header name.
+    :returns: URL if not None AND valid, None otherwise
+    """
+    url = response.headers.get(header_name)
+    try:
+        _validate(url)
+    except ValueError:
+        return None
+    else:
+        return url
+
+
+class LongRunningOperation(object):
+    """LongRunningOperation
+    Provides default logic for interpreting operation responses
+    and status updates.
+    """
+    _convert = re.compile('([a-z0-9])([A-Z])')
```

Comment:
This doesn't appear to be used anywhere?

Reply:
That's a good point, even in the current code it's not...